### PR TITLE
Development

### PR DESCRIPTION
**Pull request by nitsandaniel1**

### CODE REVIEW

File: processing.py [hunk 1]
```diff
@@ -239,15 +241,16 @@ def process_fastq(fastq_file, reference, output_dir, quality_threshold, task, ev
     os.makedirs(blast_output, exist_ok=True)
     reads_fasta_file_path = convert_fastq_to_fasta(fastq_file=fastq_file, output_dir=blast_output)
     blast_output_file = reads_fasta_file_path + ".blast"
-    run_blast(reads_fasta=reads_fasta_file_path, reference=reference, output=blast_output_file, mode=mode, task=task,
-              evalue=evalue, num_alignments=num_alignments, dust=dust, soft_masking=soft_masking, log=log,
-              perc_identity=perc_identity)
+    run_blast(reads_fasta=reads_fasta_file_path, reference=reference, output=blast_output_file, mode=mode,
```
Comment (theitush) [inline processing.py]: fix spacing please

File: processing.py [hunk 2]
```diff
@@ -239,15 +241,16 @@ def process_fastq(fastq_file, reference, output_dir, quality_threshold, task, ev
...
+                  task=task,
+                  evalue=evalue, num_alignments=num_alignments, dust=dust, soft_masking=soft_masking, log=log,
+                  perc_identity=perc_identity)
     basecall_output = os.path.join(output_dir, 'basecall')
     if not quality_threshold:
         quality_threshold = 30
     os.makedirs(basecall_output, exist_ok=True)
     basecall(blast_output_file=blast_output_file, fastq_file=fastq_file, output_dir=basecall_output,
-             quality_threshold=quality_threshold, mode=mode, reads_overlap=reads_overlap)
+                 quality_threshold=quality_threshold, mode=mode, reads_overlap=reads_overlap)
```
Comment (theitush) [inline processing.py]: fix spacing

File: runner.py [hunk 3]
```diff
@@ -74,16 +74,17 @@
...
-    except:
-        raise Exception("Processing failed. This may be due to RAM overload."
-                        "To avoid this try running with --max_memory max_available_RAM_in_MB")
+    with concurrent.futures.ProcessPoolExecutor() as executor:
+        future_tasks = {executor.submit(process_fastq, fastq_file, reference=reference_file, output_dir=processing_dir,
+                                        quality_threshold=quality_threshold, task=task, evalue=evalue, dust=dust,
+                                        num_alignments=num_alignments, soft_masking=soft_masking,
+                                        perc_identity=perc_identity, mode=mode, reads_overlap=reads_overlap): fastq_file
+                                        for fastq_file in fastq_files}
+        try:
```
Comment (theitush) [inline runner.py]: why try?

File: runner.py [hunk 4]
```diff
@@ -74,16 +74,17 @@
...
-                                    num_alignments=num_alignments, soft_masking=soft_masking,
-                                    perc_identity=perc_identity, mode=mode, reads_overlap=reads_overlap)
-    try:
-        with concurrent.futures.ProcessPoolExecutor() as executor:
-            executor.map(process_fastq_partial, fastq_files)
-    except:
-        raise Exception("Processing failed. This may be due to RAM overload."
-                        "To avoid this try running with --max_memory max_available_RAM_in_MB")
+    with concurrent.futures.ProcessPoolExecutor() as executor:
+        future_tasks = {executor.submit(process_fastq, fastq_file, reference=reference_file, output_dir=processing_dir,
```
Comment (theitush) [inline runner.py:78]: why is this a dictionary and not a list? looks like you are only using the dictionary keys anyway...

File: runner.py [hunk 5]
```diff
@@ -397,11 +398,12 @@ def create_runner_parser():
     runner(input_dir=args['input_dir'], output_dir=args['output_dir'], reference_file=args['reference_file'],
            max_basecall_iterations=int(args['max_basecall_iterations']), overlapping_reads=args['overlapping_reads'],
            quality_threshold=int(args['quality_threshold']), task=args['blast_task'], max_memory=args['max_memory'],
-           evalue=float(args['blast_evalue']), dust=args['blast_dust'], num_alignments=int(args['blast_num_alignments']),
+           evalue=float(args['blast_evalue']), dust=args['blast_dust'],
+           num_alignments=int(args['blast_num_alignments']),
            mode=args['blast_mode'], perc_identity=float(args['blast_perc_identity']), cpu_count=args['cpu_count'],
-           min_coverage=int(args['min_coverage']), db_comment=args['db_comment'], soft_masking=args['blast_soft_masking'],
+           min_coverage=int(args['min_coverage']), db_comment=args['db_comment'],
+           soft_masking=args['blast_soft_masking'],
```
Comment (theitush) [inline runner.py]: spacing

File: runner.py [hunk 6]
```diff
@@ -397,11 +398,12 @@ def create_runner_parser():
     runner(input_dir=args['input_dir'], output_dir=args['output_dir'], reference_file=args['reference_file'],
            max_basecall_iterations=int(args['max_basecall_iterations']), overlapping_reads=args['overlapping_reads'],
            quality_threshold=int(args['quality_threshold']), task=args['blast_task'], max_memory=args['max_memory'],
-           evalue=float(args['blast_evalue']), dust=args['blast_dust'], num_alignments=int(args['blast_num_alignments']),
+           evalue=float(args['blast_evalue']), dust=args['blast_dust'],
```
Comment (theitush) [inline runner.py:403]: spacing

**Review by theitush — CHANGES_REQUESTED** [review]

File: runner.py (hunk 4 above)
Comment (theitush) [reply runner.py:78]: this is the correct syntax...!

File: processing.py (hunk 1 above)
Comment (nitsandaniel1) [reply processing.py]: fixed

File: processing.py (hunk 2 above)
Comment (nitsandaniel1) [reply processing.py]: fixed

File: runner.py (hunk 3 above)
Comment (nitsandaniel1) [reply runner.py]: to catch exceptions

File: runner.py (hunk 5 above)
Comment (nitsandaniel1) [reply runner.py]: fixed

File: runner.py (hunk 6 above)
Comment (nitsandaniel1) [reply runner.py:403]: fixed

File: runner.py (hunk 3 above)
Comment (theitush) [reply runner.py]: but you always raise them after catching..... so whats the point?

File: processing.py [hunk 7]
```diff
@@ -239,15 +241,15 @@ def process_fastq(fastq_file, reference, output_dir, quality_threshold, task, ev
     os.makedirs(blast_output, exist_ok=True)
     reads_fasta_file_path = convert_fastq_to_fasta(fastq_file=fastq_file, output_dir=blast_output)
     blast_output_file = reads_fasta_file_path + ".blast"
-    run_blast(reads_fasta=reads_fasta_file_path, reference=reference, output=blast_output_file, mode=mode, task=task,
-              evalue=evalue, num_alignments=num_alignments, dust=dust, soft_masking=soft_masking, log=log,
-              perc_identity=perc_identity)
+    run_blast(reads_fasta=reads_fasta_file_path, reference=reference, output=blast_output_file, mode=mode,task=task,
+                  evalue=evalue, num_alignments=num_alignments, dust=dust, soft_masking=soft_masking, log=log,
+                  perc_identity=perc_identity)
```
Comment (theitush) [inline processing.py:246]: spacing

File: processing.py [hunk 8]
```diff
@@ -239,15 +241,15 @@ def process_fastq(fastq_file, reference, output_dir, quality_threshold, task, ev
...
     basecall_output = os.path.join(output_dir, 'basecall')
     if not quality_threshold:
         quality_threshold = 30
     os.makedirs(basecall_output, exist_ok=True)
     basecall(blast_output_file=blast_output_file, fastq_file=fastq_file, output_dir=basecall_output,
-             quality_threshold=quality_threshold, mode=mode, reads_overlap=reads_overlap)
+            quality_threshold=quality_threshold, mode=mode, reads_overlap=reads_overlap)
```
Comment (theitush) [inline processing.py:252]: spacing..

**Review by theitush — CHANGES_REQUESTED** [review]

**Pull request merged**